src/SourceGenerators/Uno.UI.Tasks/BatchMergeXaml/MergedDictionary.cs

```diff
@@ -0,0 +1,372 @@
+            {
+                foreach (XmlAttribute attribute in node.Attributes)
+                {
+                    if (attribute.Name == "TargetType")
+                    {
+                        key = attribute.Value;
+                        break;
+                    }
+                }
+            }
+
+            if (key.Length > 0)
+            {
+                // If this node has a key and a conditional-inclusion namespace, we'll attach a prefix
+                // to the key corresponding to the condition we checked in order to allow multiple such nodes
+                // with the same key to exist.
+                int indexOfContractPresent = node.NamespaceURI.IndexOf("IsApiContract");
+
+                if (indexOfContractPresent >= 0)
+                {
+                    key = node.NamespaceURI.Substring(indexOfContractPresent) + ":" + key;
+                }
+            }
+
+            return key;
+        }
+
+        private static Dictionary<string, string> standardNameDictionary = new Dictionary<string, string> {
+            { "using:Microsoft.UI.Xaml.Controls", "controls" },
+            { "using:Microsoft.UI.Xaml.Controls.Primitives", "primitives" },
+            { "using:Microsoft.UI.Xaml.Media", "media" },
+            { "http://schemas.microsoft.com/winfx/2006/xaml", "x"},
+        };
+
+        private static string _conditionalXamlPattern = @"http://schemas.microsoft.com/winfx/2006/xaml/presentation?IsApiContract(Not|)Present\(Windows.Foundation.UniversalApiContract,([0-9]+)\)";
+        private static string GetStandardNamespace(string name, string value)
+        {
+            var match = Regex.Match(value, _conditionalXamlPattern);
+            if (match.Success)
+            {
+                return "contract" + match.Groups[2].Captures[0].Value + match.Groups[1].Captures[0].Value + "Present";
+            }
+            if (standardNameDictionary.ContainsKey(value))
+            {
+                return standardNameDictionary[value];
+            }
+            return name;
+        }
+
+        private static void GenerateStandardNameSpaces(XmlDocument document, Dictionary<string, string> standardNamespaceDictionary, Dictionary<string, string> toBeReplacedDictionary)
+        {
+            foreach (XmlNode node in document.ChildNodes)
+            {
+                if (node.Name == "ResourceDictionary")
+                {
+                    foreach (XmlAttribute att in node.Attributes)
+                    {
+                        if (att.Name.StartsWith("xmlns:"))
+                        {
+                            string name = att.Name.Substring(6); // exclude "xmlns:"
+                            string stardardName = GetStandardNamespace(name, att.Value);
+                            standardNamespaceDictionary.Add(stardardName, att.Value);
+
+                            if (!name.Equals(stardardName))
```

Comment:
Codacy found an issue: [Change this call to 'name.Equals' to an overload that accepts a 'StringComparison' as a parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5930374)